src/protocol/model.rs

```diff
@@ -347,6 +350,32 @@ pub mod payment {
         pub main_key: PublicKey,
         pub recipient: PublicKey,
     }
+
+    #[derive(Debug, Clone, Serialize, Deserialize)]
+    #[cfg_attr(feature = "bindings", derive(uniffi::Record))]
+    pub struct InvoiceRequestContent {
+        pub request_id: String,
+        pub amount: u64,
+        pub currency: Currency,
+        pub current_exchange_rate: Option<ExchangeRate>,
+        pub expires_at: Timestamp,
+        pub description: Option<String>,
```

Comment:
I just remembered, here we should add an optional reference to a previous payment so that we can use this for refunds. I think the best way is to use the previous invoice in full, i would add something like

```rs
pub refund_invoice: Option<String>
```

It will be up to the app to show the user that the payment request is a refund (maybe they can add a strikethrough line on the payment activity)